File: src/chipschallenge/blockreactions/GravelTo.java
package chipschallenge.blockreactions;

import chipschallenge.Block;

/**
 *
 * @author patrik
 */
public class GravelTo extends BlockReaction {

    private GravelTo() {
    }
    private static GravelTo mInstance = null;

    public static synchronized GravelTo getInstance() {
        if (mInstance == null) {
            mInstance = new GravelTo();
        }
        return mInstance;
    }

    public void react(Block moving, Block standing) {
        // No reaction
    }

    public boolean canMove(Block moving, Block standing) {
        return moving.isChip();
    }
}

File: ChipsChallenge.Shared/Blockreactions/GravelTo.cs
namespace ChipsChallenge.Shared.Blockreactions
{
    public class GravelTo : NoSlipReaction
    {
        private static GravelTo instance;

        private GravelTo()
        {
        }

        public static GravelTo Instance
        {
            get
            {
                lock (typeof(GravelTo))
                {
                    return instance ?? (instance = new GravelTo());
                }
            }
        }

        public override void React(Block moving, Block standing)
        {
        }

        public override bool canMove(Block moving, Block standing)
        {
            return moving.Chip || moving.IsBlock();
        }
    }
}